src/core/utils/url.js

```diff
@@ -23,11 +28,60 @@ export function removeUndefinedProps(object: Object): Object {
 export function addQueryParams(
   urlString: string,
   queryParams: { [key: string]: ?string | number } = {},
+  _config: typeof config = config,
 ): string {
+  let adjustedQueryParams = { ...queryParams };
+
+  if (
+    _config.get('enableFeatureUseUtmParams') &&
+    typeof queryParams.src !== 'undefined'
+  ) {
+    adjustedQueryParams = {
+      ...queryParams,
+      // Use UTM parameters instead of `src`, according to the PRD.
+      utm_source: DEFAULT_UTM_SOURCE,
+      utm_medium: DEFAULT_UTM_MEDIUM,
+      utm_content: queryParams.src,
+      src: undefined,
+    };
+  }
+
   const urlObj = url.parse(urlString, true);
   // Clear search, since query object will only be used if search property
   // doesn't exist.
   urlObj.search = undefined;
-  urlObj.query = removeUndefinedProps({ ...urlObj.query, ...queryParams });
+  urlObj.query = removeUndefinedProps({
+    ...urlObj.query,
+    ...adjustedQueryParams,
+  });
+
   return url.format(urlObj);
 }
+
+export function getQueryParametersForAttribution(
+  location: ReactRouterLocationType,
+  _config: typeof config = config,
+): Object {
+  let params = { src: location.query.src };
+
+  if (_config.get('enableFeatureUseUtmParams')) {
+    if (typeof location.query.src !== 'undefined') {
+      params = {
+        // Use UTM parameters instead of `src`. This allows us to easily update
+        // all query parameters for attribution.
+        utm_source: DEFAULT_UTM_SOURCE,
+        utm_medium: DEFAULT_UTM_MEDIUM,
+        utm_content: location.query.src,
+      };
+    } else {
+      params = {
+        utm_campaign: location.query.utm_campaign,
+        utm_content: location.query.utm_content,
+        utm_medium: location.query.utm_medium,
+        utm_source: location.query.utm_source,
```

Comment:
Why don't we want to always pass `utm_source` and `utm_medium` as `DEFAULT_UTM_SOURCE` and `DEFAULT_UTM_MEDIUM` even if they don't exist in `location.query`? Isn't the point of this `getQueryParametersForAttribution` function to return a set of preferred utm params? What's the use case for calling this with only `utm_campaign` (for example) specified in `location.query`?

Reply:
I think we only want to pass UTM parameters that have been configured. Those params can take pretty much any values because they can be used by developers themselves. See: https://extensionworkshop-dev.allizom.org/documentation/manage/monitoring-extension-usage-statistics/#tracking-external-sources